tron/mesos.py

```diff
@@ -327,14 +333,19 @@ def submit(self, task):
             self.handle_next_event()
 
         mesos_task_id = task.get_mesos_id()
-        self.tasks[mesos_task_id] = task
-        self.runner.run(task.get_config())
-        log.info(
-            'Submitting task {} to {}'.format(
-                mesos_task_id,
-                self.mesos_address,
-            ),
-        )
+        self.tasks[mesos_task_id] = task  # handle case if task id already exists?
+        if recover:
+            task.log.info('Reconciling state for this task from Mesos')
+            task.started()
+            self.runner.check(task.get_config())
```

Comment:
since  our task ids are unique, can we treat `.run` as idempotent somehow and just call it again with the same task config? this way we'd avoid needing `.check` at all.

Reply:
Hm, are you suggesting that we always attempt to launch the task, and if the task ID has been used before, then Mesos will ignore it?

If Mesos doesn't ignore it, then how does taskproc figure out if it's submitted the task to Mesos before, or not?

Reply:
yeah, ignore me, we don't need to enforce idempotency here.